collated/test/jingyinno.md

```diff
@@ -0,0 +1,1128 @@
+    public static final String VALID_BUILDING_3 = "ERC";
+
+    public static final String INVALID_BUILDING_1 = "ERC*";
+    public static final String INVALID_BUILDING_2 = "COM1 COM2";
+    public static final String INVALID_BUILDING_3 = "Building";
+
+    public static final String VALID_UNALIAS = VALID_ALIAS_ADD;
+
+    public static final String INVALID_UNALIAS_DESC = "alias!";
+```
+###### /java/seedu/address/logic/commands/VacantCommandTest.java
+``` java
+public class VacantCommandTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void constructor_nullBuilding_throwsNullPointerException() {
+        thrown.expect(NullPointerException.class);
+        new VacantCommand(null);
+    }
+
+    @Test
+    public void execute_validBuildingRetrieval_success() throws Exception {
+        ModelStub modelStub = new ModelStubAcceptingBuilding();
+        Building validBuilding = new Building(VALID_BUILDING_1);
+
+        CommandResult commandResult = getVacantCommand(validBuilding, modelStub).execute();
+
+        assertEquals(String.format(VacantCommand.MESSAGE_SUCCESS, validBuilding), commandResult.feedbackToUser);
+    }
+
+    @Test
+    public void execute_invalidBuildingRetrieval_failure() throws Exception {
+        ModelStub modelStub = new ModelStubAcceptingBuilding();
+        Building validBuilding = new Building(INVALID_BUILDING_3);
+
+        thrown.expect(CommandException.class);
+
+        getVacantCommand(validBuilding, modelStub).execute();
+    }
+
+    @Test
+    public void equals() throws Exception {
+        Building validBuildingOne = new Building(VALID_BUILDING_1);
+        Building validBuildingTwo = new Building(VALID_BUILDING_2);
+
+        VacantCommand oneVacantCommand = new VacantCommand(validBuildingOne);
+
+        // same object -> returns true
+
+        assertTrue(validBuildingOne.equals(validBuildingOne));
+
+        // same values -> returns true
+        VacantCommand validBuildingOneCopy = new VacantCommand(validBuildingOne);
+        assertTrue(oneVacantCommand.equals(validBuildingOneCopy));
+
+        // different types -> returns false
+        assertFalse(validBuildingOne.equals(1));
+
+        // null -> returns false
+        assertFalse(validBuildingOne == null);
+
+        // different vacant -> returns false
+        assertFalse(validBuildingOne.equals(validBuildingTwo));
+    }
+
+    /**
+     * Generates a new VacantCommand with the details of the given building
+     */
+    private VacantCommand getVacantCommand(Building building, Model model) {
+        VacantCommand command = new VacantCommand(building);
+        command.setData(model, new CommandHistory(), new UndoRedoStack());
+        return command;
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
```

Comment:
Double JavaDoc here again...

In fact, if this stub is being used in multiple places, you can consider the possibility of extracting it into a file of its own